typing/types.ml

```diff
@@ -795,13 +853,16 @@ let set_level ty level =
     if ty.id <= !last_snapshot then log_change (Clevel (ty, ty.level));
     Transient_expr.set_level ty level
   end
+
 (* TODO: introduce a guard and rename it to set_higher_scope? *)
 let set_scope ty scope =
   let ty = repr ty in
-  if scope <> ty.scope then begin
-    if ty.id <= !last_snapshot then log_change (Cscope (ty, ty.scope));
+  let prev_scope = ty.scope land marks_mask in
```

Comment:
We were investigating an issue in Merlin with @voodoos and we find this line suspicious.

`prev_scope` should be set to the scope  but `land marks_mask` has the opposite effect and keep only the mark bits.
I think it should be `prev_scope land scope_mask`.

Reply:
Thanks for the pointer. And sorry for the noise, we should have checked that before :bow: